src/code/z_demo.c

```diff
@@ -1,99 +1,1650 @@
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_15F:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    func_800EB6F8(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_TEXTBOX:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    cmd = (CsCmdBase*)cutscenePtr;
+                    if (cmd->base != 0xFFFF) {
+                        Cutscene_Command_Textbox(globalCtx, csCtx, (CsCmdTextbox*)cutscenePtr);
+                    }
+                    cutscenePtr += sizeof(CsCmdTextbox);
+                }
+                break;
+
+            case CS_CMD_SCENE_TRANS_FX:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_TransitionFX(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_MOTIONBLUR:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_MotionBlur(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_GIVETATL:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_GiveTatlToPlayer(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case -2:
+                // Very useful and necessary case
+                break;
+
+            default:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_ProcessCommands.s")
+#endif
+
+void func_800ED980(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    if (gSaveContext.cutscene >= 0xFFF0) {
+        csCtx->frames++;
+        Cutscene_ProcessCommands(globalCtx, csCtx, (void*)globalCtx->csCtx.csData);
+    }
+}
+
+void func_800ED9C4(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    if (func_800EA220(globalCtx, csCtx, 0.0f)) {
+        Audio_SetCutsceneFlag(0);
```

Comment:
I've been wondering about the name `Audio_SetCutsceneFlag`. It's not really a "flag", it's more of a bool. The function itself is:
```
void Audio_SetCutsceneFlag(s8 flag) {
    sAudioCutsceneFlag = flag;
}
```
But `sAudioCutsceneFlag` is only ever used as a bool to indicate whether a cutscene is playing or not. Not sure if you had any thoughts on that. It can also take `(false)` as an argument